raiden-network/light-client
# Service to download and serve a PR artifact

Fixes #1746 , #1749 and #1511 

**Short description**
Provides a service to serve PR builds locally.

To test:
Create a `.env` file on the `serve_pr` folder, containing `CIRCLE_TOKEN=<your_token>`.

Also, create a `serve_pr/app` folder and be sure it has write and execution permissions for user:group `nobody`, as nginx's process will try to run script and write to this folder as such.

Then, after running `docker-compose up`, one should be able to go to https://localhost:8080/?branch=pull/1754 or any other PR/branch and get it downloaded and served on https://localhost:8080/pull/1754 . Notice the branch used on the first request needs to be the upstream branch: for PRs opened from external repo, it'll be `pull/<#PR>`; for PRs opened from some upstream branch, it'll be the actual branch.
To "refresh" the served files (e.g. after a new successful build), access the first (root+param) URL again.

One can also use the script directly, like:
```sh
./ci_dapp.sh branch=pull/1754 -s
```
The `-s` param will cause the build to be served on https://localhost:8080/ with a `python -m http.server`.

A README was added as per @kelsos request.

**Definition of Done**


**Steps to manually test the change (dApp)**

1.
2.
